dascore/utils/array.py

```
@@ -346,19 +348,87 @@ def _apply_op(array1, array2, operator, reversed=False):
+                raise UnitError(msg) from er
+        except TypeError:
+            # The unit registry does not implement this ufunc (or cannot
+            # hold this scalar, a bool say); numpy does, and the units are
+            # whatever they were.
+            return _apply_op(patch.data, other, operator, reversed), attrs
+        new_data = _apply_op(patch.data, other, operator, reversed)
+        if hasattr(probe, "units"):
+            return new_data, attrs.update(data_units=str(probe.units))
+        return new_data, attrs.update(data_units=None)
+
     def _apply_op_units(patch, other, operator, attrs, reversed=False):
```

Ruff (0.16.1)

[error] Function argument `reversed` is shadowing a Python builtin

(A002)